tests/foreman/api/test_classparameters.py

```diff
@@ -225,6 +226,44 @@ def test_positive_list_parameters_by_puppetclass_id(self):
 
     @run_only_on('sat')
     @tier1
+    def test_positive_list_with_non_admin_user(self):
+        """List all the parameters for specific puppet class by id.
+
+        :id: e8b140c0-5c6a-404f-870c-8ebb128830ef
+
+        :expectedresults: Parameters listed for specific Puppet class.
+
+        :BZ: 1391556
+
+        :CaseImportance: Critical
+        """
+        # Create a new user and give them minimal permissions.
+        login = gen_string('alpha')
+        password = gen_string('alpha')
+        user = entities.User(
+            login=login,
+            password=password,
+            admin=False,
+        ).create()
+        role = entities.Role().create()
+        for res_type in ['Puppetclass', 'PuppetclassLookupKey']:
+            permission = entities.Permission(resource_type=res_type).search()
+            entities.Filter(
+                permission=permission,
+                role=role
+            ).create()
+        user.role = [role]
+        user.update(['role'])
+        cfg = get_nailgun_config()
+        cfg.auth = (login, password)
+        result = entities.PuppetClass(
```

Comment:
can we add some assert that the user has only the required permission (be sure it cannot access other resources that are not in the permissions)